### PR TITLE
Fix(medusa): Totals on draft orders

### LINKED ISSUE #2742: Removing discount from cart doesn't remove line adjustments from database
### System information

Medusa version (including plugins): 1.6.5
Node.js version: 18.12.0
Database: PostgreSQL
Operating system: Windows 10
Browser (if relevant):

### Steps to reproduce the behavior

1. Create 20% off discount
2. Create draft order using 20% discount
4. Remove discount https://docs.medusajs.com/api/admin/#tag/Draft-Order/operation/PostDraftOrdersDraftOrder

If you now fetch the draft order, all the line items will still report "adjustments", even tho they have no effect.

### Expected behavior

The adjustments property for each line item should be empty after removing all discounts

### Screenshots

image

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- retrieve cart with totals on draft order operations

Fixes CORE-894
closes #2742

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]
LGTM, Just need to fix the integration :)

**Comment by josetr** [comment]
The problem that I currently have with this is that even tho all draft order endpoints now will call `retrieveWithTotals`, `get-draft-order` will do it with different arguments:

```ts
// get-draft-order
draftOrder.cart = await cartService.retrieveWithTotals(
  draftOrder.cart_id,
  {
    relations: defaultAdminDraftOrdersCartRelations,
  },
  {
    force_taxes: true,
  }
);

// create-line-item, update-line-item, delete-line-item, update-draft-order
draftOrder.cart = await cartService
      .withTransaction(manager)
      .retrieveWithTotals(draftOrder.cart_id, {
        relations: defaultAdminDraftOrdersCartRelations,
        select: defaultAdminDraftOrdersCartFields,
      })
```

**Comment by adrien2p** [comment]
> The problem that I currently have with this is that even tho all draft order endpoints now will call `retrieveWithTotals`, `get-draft-order` will do it with different arguments:
> 
> 
> 
> ```ts
> 
> // get-draft-order
> 
> draftOrder.cart = await cartService.retrieveWithTotals(
> 
>   draftOrder.cart_id,
> 
>   {
> 
>     relations: defaultAdminDraftOrdersCartRelations,
> 
>   },
> 
>   {
> 
>     force_taxes: true,
> 
>   }
> 
> );
> 
> 
> 
> // create-line-item, update-line-item, delete-line-item, update-draft-order
> 
> draftOrder.cart = await cartService
> 
>       .withTransaction(manager)
> 
>       .retrieveWithTotals(draftOrder.cart_id, {
> 
>         relations: defaultAdminDraftOrdersCartRelations,
> 
>         select: defaultAdminDraftOrdersCartFields,
> 
>       })
> 
> ```
> 
> 

Yes indeed, but the total will always be present since the relations are managed internally by an utility method.

**Comment by josetr** [comment]
@adrien2p I wasn't just talking about the totals, I just want to get a consistent response everywhere. We are forcing taxes in one place but not in the rest, so we will get different results depending on the context.

**Comment by pKorsholm** [comment]
Sorry, mb @josetr, missed the `forceTaxes`. I've set it for each of them now.

**Comment by adrien2p** [comment]
@josetr yes you are right, force taxes should be always true for orders

**Comment by olivermrbl** [comment]
> yes you are right, force taxes should be always true for orders
@adrien2p Are we sure this is the right approach? We usually default to including and only explicitly force them in places where it is absolutely required. E.g. all `/store/carts/*` endpoints don't force taxes.

**Comment by adrien2p** [comment]
> > yes you are right, force taxes should be always true for orders
> 
> 
> 
> @adrien2p Are we sure this is the right approach? We usually default to not including taxes and only explicitly force them in places where it is absolutely required. E.g. all `/store/carts/*` endpoints don't force taxes. 

Cart are not orders, @srindom confirmed me last time that all orders should have taxes

**Comment by srindom** [comment]
I think the reasonable thing to do would be to rely on the `region.automatic_taxes` setting. Here's how I think about it:

- I am a store operator that is asked to create an order for a customer
- I go through the draft order creation flow
  - When adding products, I may want to communicate to the customer what the price and total is
  - The price and totals communicated to the customer are most helpful if they are tax inclusive so having that information at hand always seems reasonable.

The only scenario I can think of where this wouldn't make sense would be in regions where customers are used to having tax-exclusive prices communicated to them. For example, in the US. In these regions, when a customer asks what the price of something is, they get the value without taxes. The tax is not accounted for until the final total is calculated. 

If we use the `automatic_taxes` setting (already done internally), this should give the correct response behavior. 

So long story short: remove `force_taxes: true` and rely on the `automatic_taxes` setting to kick in. 

---

Separately:

question(non-blocking): does marking a draft order as paid from admin ensure that taxes are applied to the order in regions where automatic_taxes = false?

**Review by fPolic — APPROVED** [review]
LGTM!

**Review by srindom — APPROVED** [review]
LGTM!
